### PR TITLE
windows fixes

### PR DESCRIPTION
**Pull request by xoviat**

### CODE REVIEW

**Comment by xoviat** [comment]
@kaadam Do you have implementations for curdir and trace lin?

**Review by pawosm-arm — COMMENTED** [review]
I always find problematic mixing `_WIN64`, `_WIN32`, `WIN32` in the same codebase. Can we do soemthing about it?

**Review by pawosm-arm — APPROVED** [review]

**Comment by kiranchandramohan** [comment]
ping @kaadam

**Review by kaadam — APPROVED** [review]
LGTM. Since this PR covers pthread fix, I will close #1045.

File: lib/scutil/pgnewfil.c [hunk 1]
```diff
@@ -19,9 +19,10 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 
-#if defined(HOST_WIN)
+#ifdef _WIN64
 #include <direct.h>
 #include <io.h>
+#include <sys/stat.h>
```
Comment (bryanpkc) [inline lib/scutil/pgnewfil.c:25]: Why does `sys/stat.h` need to be included again? It is already included above on line 19.

**Comment by xoviat** [comment]
@bryanpkc Done.

**Comment by kaadam** [comment]
@bryanpkc could you look at it again? thanks.

File: runtime/flangrti/trace_lin.c [hunk 2]
```diff
@@ -264,5 +269,91 @@ __abort_sig_init(void)
     n++;
   }
 }
+#else
```
Comment (bryanpkc) [inline runtime/flangrti/trace_lin.c:273]: Delete this blank line to maintain a consistent style.

**Comment by bryanpkc** [comment]
@xoviat Could you reorganize your changes into separate commits, i.e. one for the unification of macros, one for the `__abort_sig_init` work, one of the `pthread_atfork` fix, etc.? For each commit, please write a sufficiently clear commit message.

File: runtime/flangrti/trace_lin.c (hunk 2 above)
Comment (xoviat) [reply runtime/flangrti/trace_lin.c:273]:
```suggestion
```

**Comment by kaadam** [comment]
@bryanpkc could you look at it again? Thanks.

**Comment by bryanpkc** [comment]
@xoviat  Sorry for nit-picking, but could you please reword the commit messages to clarify what they do? Something like this:
```
[Windows] Various fixes

- Do not call pthread_atfork, which is not supported.
- Add support for exception handling.
- Include correct headers and use symbols correct for Windows (e.g. _vscprintf).
```

```
[Windows] Standardize platform macro to _WIN64
```

**Comment by bryanpkc** [comment]
@xoviat Could you address my comment above? Once the commit messages are fixed, we can merge this.

**Comment by isuruf** [comment]
I've rebased and fixed the commit messages in #1116

**Pull request closed without merging**